internal/build/lifecycle_execution.go

```diff
@@ -35,6 +38,7 @@ type LifecycleExecution struct {
 	os           string
 	mountPaths   mountPaths
 	opts         LifecycleOptions
+	TmpDir       string
```

Comment:
I don't love making this public just for tests. But the alternatives were a bit more complicated. If anyone has feedback here, I'd be glad to hear it.

Reply:
We could have a `NewLifecycleExecutionWithTmpDir` that accepts it as an arg.

Reply:
Sounds good - I just updated (in 15e52db0) the existing constructor to have a new argument (it's only called once in an internal package).